src/Faker/Name.cs

```diff
@@ -46,6 +46,15 @@ public static class Name
 		/// <returns>The randomly created first name.</returns>
 		public static string First()
 		{
+			return Resources.Name.First.Split(Config.SEPARATOR).Random().Trim();
+		}
+
+		/// <summary>
+		///   Creates a random first name, using a more aggressive caching strategy.
+		/// </summary>
+		/// <returns>The randomly created first name.</returns>
+		public static string FirstCached()
```

Comment:
Do you think that such an implementation detail should be visible to the user? Ok, it hides a compromise between performance and light memory fingerprint. But... who cares memory fingerprint?!? :-)
I would go just for the `First()` method.

Obviously, you are the boss. Last decision is yours. :-)